### DISCUSSION TITLE
How to formulate the parameters in an API call (from Privatebin)

### DISCUSSION THREAD

**Discussion by DalekDave** [YOURLS/YOURLS#3378] [Q&A] [answered]
Hello! First of all, I would like to thank you for your work on Yourls.

A long time ago, I posted about a problem calling the Yourls API from Privatebin: https://github.com/PrivateBin/PrivateBin/issues/725

Once again, I am trying to integrate Yourls and Privatebin on my instances on https://t25b.com (Privatebin) and https://t25bvxyz (Yourls).

I want to shorten the URLs that Privatebin produces, by making a passwordless API call to Yourls.

I have tried including the recommended snippet in the Privatebin config file.

The Privatebin documentation says I should use the following API call:

For YOURLS, the configured value of urlshortener should be formated like following:

```
https://yourls.org/yourls-api.php?username=admin&password=pass&action=shorturl&format=json&url=

or

https://yourls.org/yourls-api.php?signature=1002a612b4&action=shorturl&format=json&url=
```

The Yourls documentation only gives a snippet ending in ellipsis dots... 
```

Usage of the signature token

Use parameter signature in your API requests. Example:

https://yoursite/yourls-api.php?signature=1002a612b4&action=...
[#](https://docs.yourls.org/guide/advanced/passwordless-api.html#usage-of-a-time-limited-signature-token)
```

PLEASE COULD SOMEONE TELL ME WHAT CODE I SHOULD USE INSTEAD OF THE ELLIPSIS DOTS?

Thank you very much if so! :)

**Comment by dgw** [comment]
The second example from Privatebin's documentation, as you pasted, the one including the `signature` parameter. That looks like what you need to make a call with the API token, assuming their docs are up-to-date.

**Comment by DalekDave** [comment]
You can see the problem in action yourself at https://t25b.com

Trying to shorten the Privatebin generated link results in the following screenshot for me:

![Screenshot_20220706_013759](https://user-images.githubusercontent.com/64964084/177429060-3c3fe184-6136-4a74-b3d5-f636a75d6cfe.png)

**Reply by dgw** [reply]
Something at the Privatebin end needs to handle the case where the link to be shortened already exists. Alternatively, set YOURLS_UNIQUE_URLS to false in YOURLS config.

**Comment by DalekDave** [comment, marked as answer]
Hello @dgw Thank you for your input. I got the problem solved now. All information can be read in the Privatebin issue at: https://github.com/PrivateBin/PrivateBin/issues/725
